hack/test_e2e.go

```diff
@@ -0,0 +1,944 @@
+	}
+
+	cmd_string := kind + " get clusters | grep -q " + cluster_name
+	cmd := exec.Command("bash", "-c", cmd_string)
+	err = cmd.Run()
+	if err == nil {
+		cmd_string = kind + " delete cluster --name " + cluster_name
+		cmd = exec.Command("bash", "-c", cmd_string)
+		err = cmd.Run()
+		if err != nil {
+			log.Fatalf("Cannot delete cluster %s\n", cluster_name, err)
+		}
+		fmt.Printf("Previous cluster %s deleted.\n", cluster_name)
+	}
+
+	// Default Log level for all components in test clusters
+	kind_cluster_log_level := os.Getenv("KIND_CLUSTER_LOG_LEVEL")
+	if strings.TrimSpace(kind_cluster_log_level) == "" {
+		kind_cluster_log_level = "4"
+	}
+	kind_log_level := "-v3"
+	if ci_mode == true {
+		kind_log_level = "-v7"
+	}
+/*
+	// Potentially enable --logging-format
+	scheduler_extra_args := "\"v\": \"" + kind_cluster_log_level + "\""
+	controllerManager_extra_args := "\"v\": \"" + kind_cluster_log_level + "\""
+	apiServer_extra_args := "\"v\": \"" + kind_cluster_log_level + "\""
+*/
+	var CLUSTER_CIDR, SERVICE_CLUSTER_IP_RANGE string
+	
+	switch ip_family {
+	case "ipv4":
+		CLUSTER_CIDR = CLUSTER_CIDR_V4
+        SERVICE_CLUSTER_IP_RANGE = SERVICE_CLUSTER_IP_RANGE_V4
+	}
+
+	fmt.Printf("Preparing to setup %s cluster ...\n", cluster_name)
+	// Create cluster
+	// Create the config file
+	tmpl, err := template.New("kind_config_template").Parse(KIND_CONFIG_TEMPLATE)	
+	if err != nil {
+		log.Fatalf("Unable to create template %v", err)
+	}
+	kind_config_template_data := KindConfigData {
+		IpFamily:			 	ip_family,
+		ClusterCidr:			CLUSTER_CIDR,
+		ServiceClusterIpRange:	SERVICE_CLUSTER_IP_RANGE,
+	}
+
+	yamlDestPath := artifacts_directory + "/kind-config.yaml"
+	f, err := os.Create(yamlDestPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = tmpl.Execute(f, kind_config_template_data)
+	if err != nil {
+		log.Fatal(err)
+	}
+	
+	cmd_string_set := []string {
+		kind + " create cluster ",
+		"--name " + cluster_name,
+		" --image " + KINDEST_NODE_IMAGE + ":" + E2E_K8S_VERSION,
+		" --retain",
+		" --wait=1m ",
+		kind_log_level,
+		" --config=" + artifacts_directory + "/kind-config.yaml",
+	}
+	cmd_string = ""
+	for _, s := range cmd_string_set {
+		cmd_string += s
+	}
+	
+	cmd = exec.Command("bash", "-c", cmd_string)
+	err = cmd.Run()
+	if err != nil {
+		log.Fatalf("Can not create kind cluster %s\n", cluster_name, err)
+	} 
+
+	// Patch kube-proxy to set the verbosity level
+	cmd_string_set = []string {
+		kubectl + " patch -n kube-system daemonset/kube-proxy ",
+		"--type='json' ",
+		"-p='[{\"op\": \"add\", \"path\": \"/spec/template/spec/containers/0/command/-\", \"value\": \"--v=" + kind_cluster_log_level + "\" }]'",
+	}
+	cmd_string = ""
+	for _, s := range cmd_string_set {
+		cmd_string += s
+	}
+	cmd = exec.Command("bash", "-c", cmd_string)
+	err = cmd.Run()
+	if err != nil {
+		log.Fatalf("Cannot patch kube-proxy.\n", err)
+	}
+	fmt.Println("Kube-proxy patched! Guys how can I test this? To find out if it was really successful.")
```

Comment:
"Guys how can I test this? To find out if it was really successful." 😅 
I guess if the `err` is nil, the cmd executed successfully.